### PR TITLE
citra: refactor & update

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Motivation for this change
This was mainly written by @JoshuaFern, they asked me to continue their PR (#152936) as they don't have access to a linux machine right now. Review happened there as well.

I've adressed CR comments, and made some small formatting changes.

The only thing broken right now is the title bar, and the about section. They currently don't display version information, though that's not a regression. I think it should be fine for now.

cc @SuperSandro2000 you reviewed the previous PR as well.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -128,6 +128,7 @@ mapAliases ({
   cide = throw "cide was deprecated on 2019-09-11: abandoned by upstream";
   cinepaint = throw "cinepaint has been removed from nixpkgs, as it was unmaintained"; # added 2019-12-10
   cifs_utils = cifs-utils; # added 2016-08
+  citra = citra-nightly; # added 2022-01
```
Comment (SuperSandro2000) [inline pkgs/top-level/aliases.nix:131]:
```suggestion
  citra = citra-nightly; # added 2022-01-XX
```
date is missing

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by thiagokokada** [comment]
Please fix the conflicts.

File: pkgs/misc/emulators/citra/base.nix [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+in stdenv.mkDerivation {
+  inherit pname version src;
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=OFF"
+    "-DCITRA_ENABLE_COMPATIBILITY_REPORTING=ON"
+    "-DCITRA_USE_BUNDLED_FFMPEG=OFF"
+    "-DCITRA_USE_BUNDLED_QT=OFF"
+    "-DCITRA_USE_BUNDLED_SDL2=OFF"
+    "-DDYNARMIC_ENABLE_CPU_FEATURE_DETECTION=OFF"
```
Comment (SuperSamus) [inline pkgs/misc/emulators/citra/base.nix:28]: Is this necessary?
I gave a quick look at Dynarmic's source code, [it seems like](https://github.com/merryhime/dynarmic/blob/644172477eaf0d822178cb7e96c62b75caa96573/src/dynarmic/backend/x64/block_of_code.cpp#L78) CPU feature detection is done at runtime, not compile time.

**Comment by IvarWithoutBones** [comment]
My apologies, I completely forgot about this one. A friend of mine opened up an updated pull request implementing similar features: https://github.com/NixOS/nixpkgs/pull/173433

**Pull request closed without merging**